larray/core/array.py

```diff
@@ -5685,17 +5683,22 @@ def to_excel(self, filepath=None, sheet_name=None, position='A1',
                     if not os.path.isfile(filepath):
                         new_workbook = True
                     close = True
+            if overwrite_file:
```

Comment:
It seems more intuitive to me to invert the test:
```python
if new_workbook:
    overwrite_file = True
```
As an added benefit (assuming it works), in that case you can write a single
```python
wb = open_excel(filepath, overwrite_file=overwrite_file)
```
line and avoid modifying the rest of the code. I **think** it would yield the same result.